wallet-api/src/Wallet/API.hs

```diff
@@ -195,6 +201,22 @@ data WalletAPIError =
 instance FromJSON WalletAPIError
 instance ToJSON WalletAPIError
 
+newtype WalletLog = WalletLog { getWalletLog :: [Text] }
+    deriving (Eq, Ord, Show, Generic, Semigroup, Monoid)
+
+instance FromJSON WalletLog
+instance ToJSON WalletLog
+
+-- | Write some information to the log
+logMsg :: MonadWriter WalletLog m => Text -> m ()
+logMsg = tell . WalletLog . pure
+
+-- | WalletAPI with the ability to log messages and throw errors
```

Comment:
wrong type

Reply:
I meant "LoggingWallet is WalletAPI _plus_ the ability  to log messages (...)" but I'll reword it